plugins/modules/vm_snapshot_attach_disk.py

```diff
@@ -0,0 +1,209 @@
+      - UUID of the snapshot we want to use on a VM.
+    required: False
+  snapshot_type:
+    type: str
+    choices: [ user, automated ]
+    description:
+      - Specify a type that the chosen snapshot has to be.
+      - C(snapshot_type=user) - snapshot was created manually.
+      - C(snapshot_type=automated) - snapshot was created automatically through a schedule.
+    required: True
+  source_disk_slot:
+    type: int
+    description:
+      - Specify a snapshot disk slot to use on a VM.
+    required: True
+"""
+
+
+EXAMPLES = r"""
+- name: Attach a disk from an automated (scheduled) VM Snapshot to VM
+  scale_computing.hypercore.vm_snapshot_attach_disk
+    source_snapshot_uuid: ec91ce38-a795-4c0b-bc72-60f8ddba8d91
+    snapshot_type: automated
+    vm_disk_slot: 0
+  
+- name: Attach a disk from an user-made VM Snapshot to VM
+  scale_computing.hypercore.vm_snapshot_attach_disk
+    source_snapshot_uuid: ec91ce38-a795-4c0b-bc72-60f8ddba8d91
+    snapshot_type: user
+    vm_disk_slot: 0
+"""
+
+RETURN = r"""
+records:
+  description:
+    - A list of VM Snapshot records.
+  returned: success
+  type: list
+  sample:
+    - automated_trigger_timestamp: 0
+      block_count_diff_from_serial_number: 2
+      label: snap-2
+      local_retain_until_timestamp: 0
+      remote_retain_until_timestamp: 0
+      replication: true
+      snapshot_uuid: 28d6ff95-2c31-4a1a-b3d9-47535164d6de
+      timestamp: 1679397326
+      type: USER
+      vm:
+        name: snapshot-test-vm-1
+        snapshot_serial_number: 3
+        uuid: 5e50977c-14ce-450c-8a1a-bf5c0afbcf43
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import errors, arguments
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.vm_snapshot import VMSnapshot
+from ..module_utils.typed_classes import TypedVMSnapshotToAnsible
+from typing import List, Optional
+
+
+# ++++++++++++
+# Must be reviewed - not sure if that's how this should work
+# ++++++++++++
+def attach_disk(module: AnsibleModule, rest_client: RestClient):
+    vm_name = module.params["vm_name"]  # maybe this parameter is not needed? We can get vm_uuid through the specified snapshot...
+    source_snapshot_uuid = module.params["source_snapshot_uuid"]
+    # snapshot_type = module.params["snapshot_type"]
+    source_disk_slot = module.params[
+        "source_disk_slot"
+    ]  # the higher the index, the newer the disk
+
+    vm_snapshot = VMSnapshot.get_snapshots_by_query(
+        {"uuid": source_snapshot_uuid}, rest_client
+    )[0]  # there is only 1 snapshot with the specified uuid.
+    vm_uuid = vm_snapshot["vm"]["uuid"]
+
+    # TODO: add "device_snapshots" to module_utils/vm_snapshot.py
+    #  add "device_snapshots" dict to the vm_snapshot from_hypercore() method
+    #  add "vm.block_devices" dict to the vm_snapshot from_hypercore() method
+    source_disk_uuid = vm_snapshot["device_snapshots"][source_disk_slot]
```

Comment:
you will need to first filter by source_disk_type, then by source_disk_slot.

I think it is possible to have unused disk slots. Like used slots are 0,1,2, then delete disk 1, and used slots are 0,2.
Then simple '[ind]' will not work.

Reply:
You are right. It is possible. I have used a proper filtering for this.